dipy/reconst/csdeconv.py

```diff
@@ -852,25 +852,33 @@ def recursive_response(gtab, data, mask=None, sh_order=8, peak_thr=0.01,
     gtab : GradientTable
     data : ndarray
         diffusion data
-    mask : ndarray
+    mask : ndarray, optional
         mask for recursive calibration, for example a white matter mask. It has
-        shape `data.shape[0:3]` and dtype=bool.
```

Comment:
the original function does not check for the shape of the mask, nor does it enforce the bool dtype, which can lead to spectacular blowups from past experiences. That would be good to add, plus I feel like this can work on any non 3D array like this line would suggest.